impl/src/fmt/display.rs

```diff
@@ -148,14 +153,18 @@ fn expand_enum(
     let (bounds, match_arms) = e.variants.iter().try_fold(
         (Vec::new(), TokenStream::new()),
         |(mut bounds, mut arms), variant| {
-            let attrs = ContainerAttributes::parse_attrs(&variant.attrs, attr_name)?
-                .map(Spanning::into_inner)
-                .unwrap_or_default();
+            let attrs = ContainerAttributes::parse_attrs(&variant.attrs, attr_name)?;
+            if let Some(attrs) = &attrs {
+                attrs.validate_for_struct(attr_name)?;
+            };
+            let attrs = attrs.map(Spanning::into_inner).unwrap_or_default();
+
             let ident = &variant.ident;
 
             if attrs.fmt.is_none()
                 && variant.fields.is_empty()
                 && attr_name != "display"
+                && container_attrs.rename_all.is_none()
             {
                 return Err(syn::Error::new(
```

Comment:
@ModProg hmmm... I don't see why `rename_all` argument should allow implicit naming for other traits rather than `Display` only? Could you elaborate on that?

For me, the situation `assert_eq!(format!("{:o}", Enum::VariantOne), "variant_one");` seems quite awkward.

Reply:
My idea was you could use different display traits for different ways of `rename_all` if you want to.

Reply:
I have seen crates use different format traits for stuff like this in the past, but I don't mind removing the functionality. i.e. make `rename_all` only available for `Display`.

Reply:
I agree that this is weird: `assert_eq!(format!("{:o}", Enum::VariantOne), "variant_one");`

But I also think `assert_eq!(format!("{:o}", Enum::VariantOne), "VariantOne");` is weird. And as I understand that's what we do currently. I would be happier to see the `Octal` derive simply fail for unit enum variants without an `#[octal(...)]` attribute. Instead of defaulting to stringifying them in the same way as we do by default for `Display`, but that would be a breaking change. And doing another major release to fix that doesn't seem worth it.

So my vote would be, let's do whatever is easiest code wise. And let's create an issue that we add to the 3.0 milestone, to remove this weird behaviour completely then.

Reply:
@JelteF 

> And as I understand that's what we do currently.

Nope, at the moment, we [forbid implicit naming for other traits rather than `Display`](https://github.com/JelteF/derive_more/blob/v2.0.1/impl/src/fmt/display.rs#L156-L165). That's why the question was risen.